CDCgov/prime-reportstream
# Engagement/jessica/15073 louisiana validation

This PR addresses changes request by LA after their first test message

Test Steps:
1. Make sure the integration tests pass. 
2. Check the email and make sure that the fields are added that they specified other than PID 11.7 which SR does not send over and OBX.7 which we don't send over for this kind of test.


MSH.05.03 : Receiving Application Universal ID type is blank. | This field is required and should be set to 'ISO'
-- | --
– Data Analyzed: : '' |  
MSH.06.02 : Receiving Facility Universal ID is blank. | MSH.6.2 must be 'LADOH' when sending ELR data to LA
– Data Analyzed: : '' |  
MSH.06.03 : Receiving Facility Universal ID type is blank | should be set to 'ISO'
– Data Analyzed: : '' |  
PID.11.07 : REQUEST - Patient Address Type is blank. It should be populated from HL70190 table. | Are you able to provide this field?
– Data Analyzed: : '' |  
PID.11.09 : REQUEST - Patient County/Parish Code failed check. | Are you able to provide this field? Please refer to table HL7 FIPS6-4
– Data Analyzed: : 'Orleans' |  
OBX.07 : REQUIRED - No Reference Range provided. Feedback: Ex: 'Not Detected' | Qualitative Results Reference Ranges: Negative for tests resulting as Positive
– Data Analyzed: : '' | Not Detected for tests resulting as Detected Normal for tests resulting as Abnormal



## Changes
- Addressed LA's concerns with the first test message. 
- Added an integration test for them.

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
